### ISSUE TITLE
Common utilities

### ISSUE THREAD

**Issue by almarklein** [almarklein/asgish#17]
I intend to keep Asgish as simple as possible, but some tasks are common and can be abstracted as a utility. I am not sure whether I'd add any, but I might, e.g. when I find myself writing a certain piece of code in multiple places.

Examples:
* [ ] Routing
* [ ] Compression
* [ ] Request caching headers

**Comment by almarklein** [comment]
Added [make_asset_handler()](https://asgish.readthedocs.io/reference.html#asgish.utils.make_asset_handler), which is a good start, solving a common use-case (and it includes compression and caching).

As part of this work I've experimented with a middleware-like approach to add utilities like this. This is possible, and easier now that we have [normalize_response()](https://asgish.readthedocs.io/reference.html#asgish.utils.normalize_response).  A pipelined (middleware-inspired) approach might look like:
```py
async def main(request):
   resp = normalize_response(await some_handler(request))
   resp = await transform1(request, response)
   resp = await transform2(request, response)
   return resp
```

I've also played a bit with alternate syntax, e.g. `transform2(transform1(some_handler(request)))` (yes, that can work), but found it to be a bit too magical.

As for the above-mentioned utility function, it could be broken up in 2 or 3 pieces that can be stacked together, but it would leave more room for edge cases, be harder to use in practice (at least in one of my own use-cases), and actually be less efficient (e.g. the current solution pre-compresses the assets instead of doing it for each request).

In short, inside Asgish one can adopt a pipeline-like approach to handling events, but I expect more from providing utilities that solve specific problems. That does not mean that middleware *outside Asgish* can be useful, and I am very happy how [easy this is](https://github.com/almarklein/asgish/blob/master/examples/example_middleware.py).
